spec/foxsi_ospex.pro

```diff
@@ -0,0 +1,198 @@
+PRO foxsi_ospex, filename, det=det, bin=bin, erange=erange, $
+		trange=trange, center=center, radius=radius, offaxis=offaxis, $
+		cen_map=cen_map,fwhm=fwhm, let_file=let_file, type=type, $
+		split_limit=split_limit, three=three, single=single, year=year, $
+		_extra=_extra  
+
+;
+; PURPOSE: 
+;	Perform spectral fitting on FOXSI data using an optically thin thermal plasma
+;	model. This procedure produces two files: a FITS file with the ospex spectral 
+;	fitting results and an IDL .sav file with the livetime and ratio of good events 
+;	to all events.
+;
+;
+; INPUT :
+;
+;	FILENAME : 	name for output files (string)
+;
+;
+; KEYWORDS : 
+;
+;	DET :		detector number (0,1,3,4,5,6) 	
+;	BIN : 		bin width (keV)
+;	ERANGE :	energy range for spectral fit [e_start,e_end] (keV)
+;	TRANGE : 	time range [t_start,t_end] (seconds since launch)
+;	CENTER :	source center (heliospheric coordinates, arcsec)
+;	RADIUS :	radius of circular region used for spectrum (arcsec)
+;	OFFAXIS :	offaxis angle of source (arcmin) 
+;	CEN_MAP :	map center for selected target
+;	FWHM :		energy resolution (FWHM; keV)
+;	LET_FILE : 	data file to use for low energy threshold efficiency
+;	TYPE : 		detector type ('si' or 'cdte')
+;	SPLIT_LIMIT : 	threshold on each strip for the three-strip case
+;	THREE : 	If set, return the summed energy from all three strips, not just the highest value
+;	YEAR :		year of FOXSI flight (2012, 2014, or 2018)
+;
```

Comment:
Add a keyword 'STOP' to be able to stop the procedure at the end, before exiting it (see last comment), I think this is useful to look at the two plots produced during the fitting, which disappear when the procedure is done